shared/src/commonMain/kotlin/co/nimblehq/blisskmmic/data/model/TokenApiModel.kt

```diff
@@ -0,0 +1,26 @@
+package co.nimblehq.blisskmmic.domain.model
+
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class TokenApiModel(
+    @SerialName("access_token")
+    val accessToken: String,
```

Comment:
@blyscuit To avoid getting errors from server response. I suggest putting all data model attributes as `nullable`. If we define like this, we may cause potential crash if server does not return any null field.
```suggestion
    val accessToken: String? = null,
```
Please apply for the others 🙏

Reply:
Isn't the `ignoreUnknownKeys` already mitigate this crash? I'm not sure I understand this part `we may cause potential crash if server does not return any null field.` could you provide example of response from the server? If you meant a field as null then the object would not map and the catch error would trigger instead of a crash?
```

    val json = Json {
        prettyPrint = true
        isLenient = true
        ignoreUnknownKeys = true
    }
```

Reply:
@doannimble I tried decoding an object with a `null` value. It didn't crash but return a catch error as expected. 👍 This implementation is safe, for now.

<img width="1010" alt="Screen Shot 2022-11-09 at 10 33 43" src="https://user-images.githubusercontent.com/6356137/200732290-d70cf4c5-9cde-4f4d-82c9-704f3ad80a0f.png">

Reply:
@blyscuit as we're defining all the response properties are not nullable, so if we don't catch the error, it crashes. But we're doing try/catch, so we will have the exception.
I suppose somehow, the field `createdAt`  from server is null. Actually, we don't use this field for login verification, but we will receive the error instead of login successfully. It is not good enough, ideally, we should not rely on the server's response. We should control the flow imo. What do you thik?

Reply:
I think if the object from the API is not correctly return and the app is unable to map the result then an error should be thrown. If we don't use a field then we can just leave it out of the mapping, from the example, if we don't use `createdAt` then we can just NOT define `createdAt` in the model.
For this Token model, I assume all parts declared are useful and should exist, or else error should be throw.
I think we should use `optional` for model with backend specification that some value can be null and is part of the business logic.

Reply:
I agree with your point @blyscuit . I give you another usecase in WORPT project that I used to work, when the backend team decide to change the attribute from mandatory to optional, we don't need to rebuild the app. As in developing, the backend team can change frequently, especially when we load a large bundle, we should not take more effort to change. How do you feed about this?

Reply:
Good point, I think it's depends on how much streamline the planning and documentation of the project. The same idea apply to changing one field from API to optional and the frontend can except optional, but one of the platforms(or all) catch the error differently than the expected outcome that the business side wants (show a placeholder vs move user screen back), then we still have to rework on the one platform or both. This is because the API is an unknown variable at the beginning so we have to rework. For other project where the API is known, we can concise the code more to fit the business case, the Clean pattern we are using should be easy enough to modify the logic with the least effort. For this project, I think it lies in the known area (the API will not change for a long time) but it varies from project to project.

Reply:
Okay, it's not a big due, I think go ahead with the current one 👌